tests/displays/cavity_display/frontend/test_decoder.py

```diff
@@ -0,0 +1,246 @@
+import unittest
+from unittest.mock import patch, mock_open
+from PyQt5.QtWidgets import QApplication, QSizePolicy, QAbstractScrollArea
+from PyQt5.QtCore import Qt
+from displays.cavity_display.frontend.decoder import DecoderDisplay, Row, rows
+
+
+
+class TestDecoderDisplay(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Need to create a QApplication instance before running any PyQt tests
+        cls.app = QApplication([])
```

Comment:
Ok so I learned the hard way that this can cause problems if a new QApp is created in multiple test files. A quick fix is to have it inside of a `if not QApplication.instance():` check